examples/fault-injection/envoy.yaml

```diff
@@ -36,6 +36,11 @@ static_resources:
                 percentage:
                   numerator: 0
                   denominator: HUNDRED
+              delay:
```

Comment:
This is a great find, thanks for tracking it down!

I do wonder if we should tweak the fault filter to delay / abort if there's a runtime override even if the proto doesn't include the config.  Even though it's pretty clearly documented, I can imagine it confusing folks.
cc @mattklein123 

either way it shouldn't block this PR, more musing if we should file a tracking issue.

Reply:
I don't recall the history of why this is implemented the way it is. I agree it's potentially confusing. I think the idea is we want the user to opt-in explicitly to the config they want but I don't feel super strongly about it. @Augustyniak any thoughts here?

Reply:
I think explicitness here is good since we support injecting faults via different methods including [HTTP headers](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/fault_filter#controlling-fault-injection-via-http-headers) and we want these faults to be injected only if our Envoy is explicitly configured to accept them.

- Without Envoy requiring people to opt-in with this config, it may be confusing as to what kind of faults a given Envoy may potentially inject - the ones controlled via runtime or via HTTP headers. With config in, it becomes clear what’s supported.
- If we decide that we don’t want to require a config for faults controlled via runtime we should probably do the same for faults controlled with HTTP headers which is risky because of security implications.

For all of the aforementioned reasons, I would leave the presence of the config as the required step of `FaultFilter`’s setup.

For reference, an example of a config that supports delay faults via HTTP headers:
```
delay:
  header_delay: {}
  percentage:
  numerator: 100
```